### ISSUE TITLE
ETA On Compare Models

### ISSUE THREAD

**Issue by rajatscibi** [pycaret/pycaret#2428]
**Is your feature request related to a problem? Please describe.**
Often when dataset is large or it has more number of features, the compare_models() functions takes a lot of time, sometimes hours on a regular 4GB RAM, SSD storage laptop without GPU. It's often not clear how much time it will take which makes it bit difficult to plan certain things and one may feel stuck. 

**Describe the solution you'd like**
If we can have an ETA (estimated time remaining) on compare_models(), it will be very helpful. 

**Describe alternatives you've considered**

**Additional context**

**Comment by ngupta23** [comment]
I think this is not possible right now since there is no way to probe the underlying models on the amount of time remaining.

@Yard1 - any comments on this request?

**Comment by rajatscibi** [comment]
Found this [here](https://stackoverflow.com/questions/22210768/getting-training-time-in-scikit)



```
import time
from sklearn.neural_network import MLPClassifier
from sklearn.datasets import load_iris


model = MLPClassifier()
X, y = load_iris(return_X_y=True)
start = time.time()
model.fit(X, y)
stop = time.time()
print(f"Training time: {stop - start}s")
# prints: Training time: 0.20307230949401855s

```

**Comment by rajatscibi** [comment]
There is something called as scitime as mentioned [here](https://www.freecodecamp.org/news/two-hours-later-and-still-running-how-to-keep-your-sklearn-fit-under-control-cc603dc1283b/)

> Scitime is a python package requiring at least python 3.6 with [pandas](https://github.com/pandas-dev/pandas), [scikit-learn](https://github.com/scikit-learn/scikit-learn), [psutil](https://github.com/giampaolo/psutil) and [joblib](https://github.com/joblib/joblib) dependencies. You will find the Scitime repo [here](https://github.com/nathan-toubiana/scitime).
> 
> The main function in this package is called “time”. Given a matrix vector X, the estimated vector Y along with the Scikit Learn model of your choice, time will output both the estimated time and its confidence interval.

**Comment by ngupta23** [comment]
This seems to be quite an invasive method to compute run time. You are essentially building models around the model that you really want to build. Also, not sure this applies to all sktime functionality such as `cross_validate`, [`GridSearchCV`](http://scikit-learn.org/stable/modules/generated/sklearn.model_selection.GridSearchCV.html), etc. 

Would prefer a method that is more tightly coupled with sklearn functionality. Also, would like to hear the thoughts of the other core developers as well.

**Comment by rajatscibi** [comment]
Understandable. Lets keep this feature request open for now for someone else to come up with some idea on its execution.

**Comment by Yard1** [comment]
Yeah, none of the available solution would work here (or indeed, for all of our models).

**Comment by rajatscibi** [comment]
Is it possible to estimate once the training has started and some time has elapsed. For example 10% got completed in 60 seconds, so approximately ETA to reach 100% can be around 600 seconds more or less?

**Comment by rajatscibi** [comment]
> 

Ill put this on sklearn repo as well and let see what they have to say about it. Sharing on stackoverflow as well for computer science community to suggest something.

**Comment by ngupta23** [comment]
> Is it possible to estimate once the training has started and some time has elapsed. For example 10% got completed in 60 seconds, so approximately ETA to reach 100% can be around 600 seconds more or less?

This is not true since training is not a linear process.

**Comment by ngupta23** [comment]
> > 
> 
> Ill put this on sklearn repo as well and let see what they have to say about it. Sharing on stackoverflow as well for computer science community to suggest something.

This may be the best path forward. This should be handled in the sklearn repo itself rather than as a wrapper around the models outside sklearn. That would be the most sustainable way to do this (if it is even possible).

**Comment by rajatscibi** [comment]
Reply from scikitlearn:

> This is a recurrent feature request that will be resolved by the ongoing work on a callback API (#22000)
> 
> I'm closing this issue because it is a duplicate of #10973 and #78.

[Link to post](https://github.com/scikit-learn/scikit-learn/issues/23156)
